src/components/auth/signIn/email/SignInModalContent.tsx

```diff
@@ -0,0 +1,290 @@
+      rules={[
+        {
+          pattern: RegexValidations.ValidEmail,
+          message: 'Please enter a valid email address.',
+          validateTrigger: ['onBlur'],
+        },
+      ]}
+      validateStatus={isStr(error) ? 'error' : undefined}
+    >
+      <Input
+        autoFocus
+        required
+        type='email'
+        placeholder='Email'
+        onChange={e => {
+          handleAfterInput(e.target.value)
+        }}
+        onBlur={e => {
+          handleAfterInput(e.target.value)
+        }}
+      />
+    </Form.Item>
+  )
+}
+
+export const PasswordInput = ({ error, isError, form }: InputProps) => {
+  const validatePassword = () => {
+    form.validateFields(['password'])
+  }
+  const debouncedValidateField = useCallback(
+    debounceFunction(validatePassword, VALIDATION_DELAY),
+    [],
+  )
+
+  const handleAfterInput = (value: string) => {
+    form.setFieldsValue({ [fieldName('password')]: value.trim() })
+    debouncedValidateField()
+  }
+
+  return (
+    <Form.Item
+      name={fieldName('password')}
+      className={clsx(styles.BaseFormItem, isError && styles.BaseFormItemError)}
+      rules={[
+        {
+          min: 8,
+          pattern: RegexValidations.ValidPassword,
+          message:
+            'Your password should contain at least 8 characters, with at least one letter and one number.',
+          validateTrigger: ['onBlur'],
+        },
+      ]}
+      help={error}
+      validateStatus={isStr(error) ? 'error' : undefined}
+    >
+      <Input
+        required
+        type='password'
+        placeholder='Password'
+        onChange={e => {
+          handleAfterInput(e.target.value)
+        }}
+        onBlur={e => {
+          handleAfterInput(e.target.value)
+        }}
+      />
+    </Form.Item>
+  )
+}
+
+const SignInModalContent = ({ setCurrentStep, onSignInSuccess }: Props) => {
+  const {
+    state: { email },
+    setEmail,
+    setPassword,
+  } = useAuth()
+
+  const { isValidEmail, isValidPassword } = useFormValidation()
+
+  const { setSignerTokensByAddress } = useSignerExternalStorage()
+
+  const [form] = Form.useForm()
+
+  const [isFormValid, setIsFormValid] = useState(false)
+  const [error, setError] = useState<string | null>(null)
+
+  const [token, setToken] = useState<string | undefined>()
+  const [captchaReady, setCaptchaReady] = useState(false)
+  const [loading, setLoading] = useState(false)
+
+  const hCaptchaRef = useRef(null)
+
+  const onExpire = () => {
+    setError('hCaptcha Token Expired')
+  }
+
+  const onError = (err: any) => {
+    setError(`hCaptcha Error: ${err}`)
+  }
+
+  const loadCaptcha = () => {
+    // this reaches out to the hCaptcha JS API and runs the
+    // execute function on it. you can use other functions as
+    // documented here:
+    // https://docs.hcaptcha.com/configuration#jsapi
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    hCaptchaRef.current?.execute()
```

Comment:
SignIn and SignUp Modal have copy pasted code

Reply:
Yes, all the code are related to hCaptcha and I've tried extracting them, but it makes the hCaptcha doesn't work.

Will try again with another approach.